table.go

```diff
@@ -448,7 +463,26 @@ func (t *Table) ArrowSchema(
 		case <-ctx.Done():
 			return nil, ctx.Err()
 		default:
-			record, err := pqarrow.ParquetRowGroupToArrowRecord(ctx, pool, rg, projections, filterExpr, distinctColumns)
+			schema, err := pqarrow.ParquetRowGroupToArrowSchema(
+				ctx,
+				pool,
+				rg,
+				projections,
+				filterExpr,
+				distinctColumns,
+			)
+			if err != nil {
+				return nil, err
+			}
+
+			record, err := pqarrow.ParquetRowGroupToArrowRecord(
```

Comment:
why is just using the above schema not enough here?

Reply:
Good point. It is and was an oversight, I guess. Updated.